frame/merkle-mountain-range/src/tests.rs

```diff
@@ -241,6 +241,17 @@ fn should_generate_proofs_correctly() {
 				crate::Pallet::<Test>::generate_batch_proof(vec![leaf_index]).unwrap()
 			})
 			.collect::<Vec<_>>();
+		// when generate historical proofs for all leaves
+		let historical_proofs = (0_u64..crate::NumberOfLeaves::<Test>::get())
+			.into_iter()
+			.map(|leaf_index| {
+				crate::Pallet::<Test>::generate_historical_batch_proof(
+					vec![leaf_index],
+					leaf_index + 1,
```

Comment:
I'd also add test, where `leaf_index`, `best_known_block` and actual `best_block` are all different. E.g. in this test (iiuc), `best_block` is set to `7` and `best_known_block` is always matching (+1) the `leaf_index`. Would be good to have something like `{ leaf_index = 2, best_known_block = 5, best_block = 7 }`. It is optional, but it'll be used this way in the real world, so good to have it as an illustration. wdyt?

Reply:
It's a good idea. Done.